public/app/plugins/datasource/loki/result_transformer.ts

```diff
@@ -290,9 +290,12 @@ export function createMetricLabel(labelData: { [key: string]: string }, options?
 }
 
 function getOriginalMetricName(labelData: { [key: string]: string }) {
-  const metricName = labelData.__name__ || '';
-  delete labelData.__name__;
-  const labelPart = Object.entries(labelData)
+  // we do not want to mutate the origianl labels-structure,
+  // so we make a copy
+  const labels = { ...labelData };
+  const metricName = labels.__name__ || '';
+  delete labels.__name__;
+  const labelPart = Object.entries(labels)
```

Comment:
```suggestion
  const labelPart = Object.entries(labels)
```

Loki doesn't use `__name__` label for metrics, so this is always `''` and therefore it can be removed.